### PR TITLE
[My Registrations][ENG-528] Navbar links for My Registration page

### PR DESCRIPTION
**Pull request by adlius**
- Ticket: [ENG-528]
- Feature flag: n/a

## Purpose

Adding navbar link to "My Registrations" page. Also change the links into icons for mobile view.

## Summary of Changes

Screenshot from 2021-03-23 13-23-34
Screenshot from 2021-03-23 13-23-48
Screenshot from 2021-03-23 13-24-24
Screenshot from 2021-03-23 13-24-08

## Side Effects
For mobile view, the "OSF Registries" in the navbar is removed. Otherwise there wouldn't be enough space for the icons.

[ENG-528]: https://openscience.atlassian.net/browse/ENG-528

### CODE REVIEW

File: lib/osf-components/app/components/provider-metadata-editable/template.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default } from 'osf-components/components/provider-metadata-editable/template';
```
Comment (brianjgeiger) [inline lib/osf-components/app/components/provider-metadata-editable/template.js:1]: This doesn't look like it belongs here.

File: translations/en-us.yml [hunk 2]
```diff
@@ -269,6 +269,7 @@ navbar:
     # add_a_preprint: 'Add a {preprintWords.preprint}'
     add_registration: 'Add New'
     moderation: 'Moderation'
+    my_registration: 'My Registration'
```
Comment (brianjgeiger) [inline translations/en-us.yml:272]: Should this be plural?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
A couple of small things, plus a failing test.

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Just a little copy-pasta that I see

**Review by brianjgeiger — DISMISSED** [review]

**Comment by brianjgeiger** [comment]
I'm not sure what's going on with the tests here, though.

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 3]
```diff
@@ -62,24 +62,58 @@
...
+                    local-class='SecondaryNavLinkButton HideOnMobile'
                     @route='registries.branded.moderation'
                     @models={{array this.provider.id}}
                 >
                     {{t 'navbar.moderation'}}
                 </nav.links.secondary>
+                <nav.links.secondary
+                    data-test-my-reviewing-button
+                    data-analytics-name='Go to moderation view'
+                    local-class='SecondaryNavLinkButton OnlyOnMobile'
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:74]: Needs an aria-label attribute as there is no "discernible text" to indicate where this link point to.

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 4]
```diff
@@ -62,24 +62,58 @@
...
                     @route='registries.branded.new'
                     @models={{array this.providerId}}
                 >
                     {{t 'navbar.add_registration'}}
                 </nav.links.secondary>
+                <nav.links.secondary
+                    data-test-add-new-button
+                    data-analytics-name='Create new registration'
+                    local-class='SecondaryNavLinkButton OnlyOnMobile'
+                    @route='registries.branded.new'
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:95]: Ditto ☝🏽

File: lib/registries/addon/components/registries-navbar/template.hbs [hunk 5]
```diff
@@ -62,24 +62,58 @@
...
+                data-analytics-name='Go to my registrations'
+                local-class='SecondaryNavLinkButton HideOnMobile'
+                @route='registries.my-registrations'
+            >
+                {{t 'navbar.my_registrations'}}
+            </nav.links.secondary>
+            <nav.links.secondary
+                data-test-add-new-button
+                data-analytics-name='Go to my registrations'
+                local-class='SecondaryNavLinkButton OnlyOnMobile'
```
Comment (fabmiz) [inline lib/registries/addon/components/registries-navbar/template.hbs:112]: Ditto ☝🏽

**Review by fabmiz — CHANGES_REQUESTED** [review]
Please fix the tests and address accessibility concerns

**Review by futa-ikeda — APPROVED** [review]
🚀

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**